src/ICal/ICal.php

```diff
@@ -217,10 +217,13 @@ public function __construct($files = false, array $options = array())
             $files = is_array($files) ? $files : array($files);
 
             foreach ($files as $file) {
-                if (file_exists($file)) {
+                if ($this->isUrlOrFile($file)) {
                     $lines = file($file, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
+                    if ($lines === false) {
+                        throw new \Exception("The URL '$file' does not exists.");
+                    }
                 } else {
-                    $lines = is_array($file) ? $file : array($file);
+                    throw new \Exception("The URL or file '$file' is not valid.");
```

Comment:
This is a breaking change that removes the functionality of passing in an iCal string to the constructor